overlord/snapstate/handlers.go

```diff
@@ -1164,23 +1164,33 @@ func maybeRestart(t *state.Task, info *snap.Info) {
 		return
 	}
 
+	restartMsg := daemonRestartMessage(st, typ)
+	if restartMsg == "" {
+		// no message -> no restart
+		return
+	}
+
+	t.Logf(restartMsg)
+	st.RequestRestart(state.RestartDaemon)
+}
+
+func daemonRestartMessage(st *state.State, typ snap.Type) string {
```

Comment:
I have mixed feelings on this change, but no strong -1,  but I think calling this daemoRestartReason would make it a bit clearer

Reply:
...Reason works so much better!